jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/ImageName.java

```diff
@@ -220,7 +232,13 @@ public String getFullName(String optionalRegistry) {
      * @return user part or <code>null</code> if no user is present in the name
      */
     public String getUser() {
```

Comment:
Not following completely. Is `user` as an entity relevant for the image name?
The PR says `Remove user field from ImageName`, however we just seem to  be delaying its computation

Reply:
I don't think it's relevant for image name. There are a few places in codebase where we seem to be checking for image user:

- AuthConfigFactory: while fetching credentials from settings.xml, we check for server attributes whether they have same id as registry name or ${registry}/${user}
- KubernetesResourceUtil: while extracting container name, if alias is not provided then either ${user}-${project.artifactId} or ${project.groupId}-${project.artifactId} is used
TriggersAnnotationEnricher
- BaseGenerator: while setting JKubeBuildStrategy.SourceStrategy.namespace attribute
- TriggersAnnotationEnricher: to decide whether add triggers annotation or not

I just kept this method for these places to work.

Reply:
OK, would need to check those methods. But is it relevant to call this `user`? Meaning, is this acting as a user (name) in any of the mentioned cases? If not, maybe we can rename it to call it something more relevant, else it will continue to cause confusion

Reply:
No it still seems to be unrelated to the image name.
I'll try to check out exactly how the method is consumed and see if I can propose any alternatives.